apps/trading/pages/_app.page.tsx

```diff
@@ -7,27 +8,49 @@ import {
   VegaManageDialog,
   VegaWalletProvider,
 } from '@vegaprotocol/wallet';
-import { EnvironmentProvider } from '@vegaprotocol/environment';
+import {
+  EnvironmentProvider,
+  useEnvironment,
+  Networks,
+} from '@vegaprotocol/environment';
 import { Connectors } from '../lib/vega-connectors';
 import { ThemeSwitcher } from '@vegaprotocol/ui-toolkit';
 import { AppLoader } from '../components/app-loader';
 import { VegaWalletConnectButton } from '../components/vega-wallet-connect-button';
+import { RiskNoticeDialog } from '../components/risk-notice-dialog';
 import './styles.css';
 import { useGlobalStore } from '../stores';
 import {
   AssetDetailsDialog,
   useAssetDetailsDialogStore,
 } from '@vegaprotocol/market-list';
+import { LocalStorage } from '@vegaprotocol/react-helpers';
+
+const RISK_ACCEPTED_KEY = 'vega-risk-accepted';
 
 function AppBody({ Component, pageProps }: AppProps) {
   const store = useGlobalStore();
+  const { VEGA_ENV } = useEnvironment();
   const {
     isAssetDetailsDialogOpen,
     assetDetailsDialogSymbol,
     setAssetDetailsDialogOpen,
   } = useAssetDetailsDialogStore();
   const [theme, toggleTheme] = useThemeSwitcher();
 
+  useEffect(() => {
```

Comment:
chore (blocking): can we add some tests for this as well please?

Reply:
totally hear you.  i gave this a go, the amount of things i need to wire in just to display a page is pretty ridiculous. not sure if this simple if statement is worth the mock party.

Reply:
@notbot00 how about moving the useEffect and required imports to the RiskNoticeDialog itself? It should be much easier to unit test then and keeps _app.page.tsx cleaner

Reply:
i really felt repelled by the idea at first, no idea why, my my OCD with the consistency with the other app dialogs. this is actually a super good solution i think, thanks @mattrussell36 🙏  added some tests finally